File: api-asn/src/tr/gov/tubitak/uekae/esya/api/asn/cms/EEncapsulatedContentInfo.java
package tr.gov.tubitak.uekae.esya.api.asn.cms;

import java.io.InputStream;

import tr.gov.tubitak.uekae.esya.api.asn.BaseASNWrapper;
import tr.gov.tubitak.uekae.esya.api.common.ESYAException;
import tr.gov.tubitak.uekae.esya.asn.cms.EncapsulatedContentInfo;

import com.objsys.asn1j.runtime.Asn1ObjectIdentifier;
import com.objsys.asn1j.runtime.Asn1OctetString;

public class EEncapsulatedContentInfo extends BaseASNWrapper<EncapsulatedContentInfo>
{
	public EEncapsulatedContentInfo(EncapsulatedContentInfo aObject)
	{
		super(aObject);
	}
	
	public EEncapsulatedContentInfo(Asn1ObjectIdentifier aContentType,Asn1OctetString aContent)
	{
		super(new EncapsulatedContentInfo());
		setContentType(aContentType);
		setContent(aContent);
	}
	
	public EEncapsulatedContentInfo(InputStream aIns)
	throws ESYAException
	{
		super(aIns,new EncapsulatedContentInfo());
	}
	
	
	public Asn1ObjectIdentifier getContentType()
	{
		return mObject.eContentType;
	}
	
	public void setContentType(Asn1ObjectIdentifier aContentType)
	{
		mObject.eContentType = aContentType;
	}
	
	public byte[] getContent()
	{
		if(mObject.eContent!=null)
			return mObject.eContent.value;
		
		return null;
	}
	
	public void setContent(Asn1OctetString aContent)
	{
		mObject.eContent = aContent;
	}
	
}

File: api-asn/csharp/src/api/asn/cms/EEncapsulatedContentInfo.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using tr.gov.tubitak.uekae.esya.asn.cms;
using Com.Objsys.Asn1.Runtime;
using System.IO;
namespace tr.gov.tubitak.uekae.esya.api.asn.cms
{
    public class EEncapsulatedContentInfo : BaseASNWrapper<EncapsulatedContentInfo>
    {
        public EEncapsulatedContentInfo(EncapsulatedContentInfo aObject)
            : base(aObject)
        {
        }

        public EEncapsulatedContentInfo(Asn1ObjectIdentifier aContentType, Asn1OctetString aContent)
            : base(new EncapsulatedContentInfo())
        {
            setContentType(aContentType);
            setContent(aContent);
        }

        public EEncapsulatedContentInfo(Stream aIns)
            : base(aIns, new EncapsulatedContentInfo())
        {
        }


        public Asn1ObjectIdentifier getContentType()
        {
            return mObject.eContentType;
        }

        public void setContentType(Asn1ObjectIdentifier aContentType)
        {
            mObject.eContentType = aContentType;
        }

        public byte[] getContent()
        {
            if (mObject.eContent != null)
                return mObject.eContent.mValue;

            return null;
        }

        public void setContent(Asn1OctetString aContent)
        {
            mObject.eContent = aContent;
        }
    }
}
